subgraphs/inverse-finance/src/common/getters.ts

```diff
@@ -0,0 +1,398 @@
+    market.inputTokenPriceUSD = BIGDECIMAL_ZERO;
+    market.outputTokenSupply = BIGINT_ZERO;
+    market.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+    market.exchangeRate = BIGDECIMAL_ZERO;
+    market.rewardTokenEmissionsAmount = [BIGINT_ZERO, BIGINT_ZERO];
+    market.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+    //market.snapshots - derived and don't need to be initialized
+    //market.deposits
+    //market.withdraws
+    //market.borrows
+    //market.repays
+    //market.liquidates
+
+    market.createdTimestamp = event.block.timestamp;
+    market.createdBlockNumber = event.block.number;
+    market.save();
+  }
+  return market;
+}
+
+export function getOrCreateMarketDailySnapshot(event: ethereum.Event): MarketDailySnapshot {
+  let days = event.block.timestamp.toI64() / SECONDS_PER_DAY;
+  let daysStr: string = days.toString();
+  let marketId = event.address.toHexString();
+  let id = prefixID(marketId, daysStr);
+
+  let marketMetrics = MarketDailySnapshot.load(id);
+  if (marketMetrics == null) {
+    marketMetrics = new MarketDailySnapshot(id);
+
+    marketMetrics.protocol = FACTORY_ADDRESS;
+    marketMetrics.market = marketId;
+    marketMetrics.blockNumber = event.block.number;
+    marketMetrics.timestamp = event.block.timestamp;
+    marketMetrics.rates = [
+      prefixID(marketId, InterestRateSide.LENDER, InterestRateType.VARIABLE),
+      prefixID(marketId, InterestRateSide.BORROWER, InterestRateType.VARIABLE),
+    ];
+    marketMetrics.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    marketMetrics.totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+    marketMetrics.dailyDepositUSD = BIGDECIMAL_ZERO;
+    marketMetrics.cumulativeDepositUSD = BIGDECIMAL_ZERO;
+    marketMetrics.totalBorrowBalanceUSD = BIGDECIMAL_ZERO;
+    marketMetrics.dailyBorrowUSD = BIGDECIMAL_ZERO;
+    marketMetrics.cumulativeBorrowUSD = BIGDECIMAL_ZERO;
+    marketMetrics.dailyLiquidateUSD = BIGDECIMAL_ZERO;
+    marketMetrics.cumulativeLiquidateUSD = BIGDECIMAL_ZERO;
+
+    marketMetrics.inputTokenBalance = BIGINT_ZERO;
+    marketMetrics.inputTokenPriceUSD = BIGDECIMAL_ZERO;
+    marketMetrics.outputTokenSupply = BIGINT_ZERO;
+    marketMetrics.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+    marketMetrics.exchangeRate = BIGDECIMAL_ZERO;
+
+    marketMetrics.rewardTokenEmissionsAmount = [BIGINT_ZERO];
+    marketMetrics.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO];
+
+    marketMetrics.save();
+  }
+
+  return marketMetrics;
+}
+
+export function getOrCreateMarketHourlySnapshot(event: ethereum.Event): MarketHourlySnapshot {
+  let days = event.block.timestamp.toI64() / SECONDS_PER_DAY;
+  let daysStr: string = days.toString();
+  let secondsPastMidnight = event.block.timestamp.toI64() % SECONDS_PER_DAY;
+  // HH: hour of the day
+  let hours = secondsPastMidnight / SECONDS_PER_HOUR;
+  let hoursStr = hours.toString();
+
+  let marketId = event.address.toHexString();
+  let id = prefixID(marketId, daysStr, hoursStr);
```

Comment:
The ID for hour for schema 1.2.1 has been changed to just a single number. See comments in the schema